tests/samplers_tests/tpe_tests/test_multi_objective_sampler.py

```diff
@@ -244,58 +244,39 @@ def test_multi_objective_sample_independent_misc_arguments() -> None:
+    log: bool, step: Optional[float]
+) -> None:
+    # Prepare sample from float distribution for checking other distributions.
     study = optuna.create_study(directions=["minimize", "maximize"])
     random.seed(128)
-    past_trials = [frozen_trial_factory(i, [random.random(), random.random()]) for i in range(16)]
+    float_dist = optuna.distributions.FloatDistribution(1.0, 100.0, log=log, step=step)
 
-    uni_dist = optuna.distributions.FloatDistribution(1.0, 100.0)
-    trial = frozen_trial_factory(16, [0, 0])
-    sampler = TPESampler(seed=0)
-    attrs = MockSystemAttr()
-    with patch.object(study._storage, "get_all_trials", return_value=past_trials), patch.object(
-        study._storage, "set_trial_system_attr", side_effect=attrs.set_trial_system_attr
-    ), patch.object(study._storage, "get_trial", return_value=trial), patch(
-        "optuna.trial.Trial.system_attrs", new_callable=PropertyMock
-    ) as mock1, patch(
-        "optuna.trial.FrozenTrial.system_attrs",
-        new_callable=PropertyMock,
-    ) as mock2:
-        mock1.return_value = attrs.value
-        mock2.return_value = attrs.value
-        uniform_suggestion = sampler.sample_independent(study, trial, "param-a", uni_dist)
-    assert 1.0 <= uniform_suggestion < 100.0
-
-
-def test_multi_objective_sample_independent_log_uniform_distributions() -> None:
-    """Prepare sample from uniform distribution for cheking other distributions."""
+    if float_dist.step:
 
-    study = optuna.create_study(directions=["minimize", "maximize"])
-    random.seed(128)
+        def value_fn(number: int) -> float:
+            return int(random.random() * 1000) * 0.1
 
-    uni_dist = optuna.distributions.FloatDistribution(1.0, 100.0)
-    past_trials = [frozen_trial_factory(i, [random.random(), random.random()]) for i in range(16)]
-    trial = frozen_trial_factory(16, [0, 0])
-    sampler = TPESampler(seed=0)
-    attrs = MockSystemAttr()
-    with patch.object(study._storage, "get_all_trials", return_value=past_trials), patch.object(
-        study._storage, "set_trial_system_attr", side_effect=attrs.set_trial_system_attr
-    ), patch.object(study._storage, "get_trial", return_value=trial), patch(
-        "optuna.trial.Trial.system_attrs", new_callable=PropertyMock
-    ) as mock1, patch(
-        "optuna.trial.FrozenTrial.system_attrs",
-        new_callable=PropertyMock,
-    ) as mock2:
-        mock1.return_value = attrs.value
-        mock2.return_value = attrs.value
-        uniform_suggestion = sampler.sample_independent(study, trial, "param-a", uni_dist)
+        past_trials = [
+            frozen_trial_factory(
+                i, [random.random(), random.random()], dist=float_dist, value_fn=value_fn
+            )
+            for i in range(16)
+        ]
+    else:
+        past_trials = [
+            frozen_trial_factory(i, [random.random(), random.random()], dist=float_dist)
+            for i in range(16)
+        ]
```

Comment:
I'm wondering if we can simplify it based on the current approach like following:

```python
    if float_dist.step:
        value_fn: Optional[Callable[[int], int]] = lambda number: int(random.random() * 1000) * 0.1
    else:
        value_fn = None

    past_trials = [
        frozen_trial_factory(i, [random.random(), random.random()], dist=float_dist, value_fn=value_fn)
        for i in range(16)
    ]
```

What do you think?

Reply:
Agreed, I couldn't figure out how to pass mypy check and ended up like that. Thank you very much for letting me know the solution with type annotation!

I think `Optional[Callable[[int], float]]` should be used but please let me know if otherwise.

Reply:
> I think Optional[Callable[[int], float]] should be used but please let me know if otherwise.

You're totally right, my annotation was wrong. :bow: